### PR TITLE
When creating an account with MAS, change the `prompt` parameter to `create`

### LINKED ISSUE #3627: Send the user to oidc register page when creating account
#### What would you like to do?

Send the user directly to the register page on an oidc enabled homeserver when going through the create account flow

#### Why would you like to do it?

So the user doesn't have to take an extra step to create an account after already expressing their intent to create an account

#### How would you like to achieve it?

Add `prompt=create` to the authorization url

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by bmarty**
## Content

When user is creating account, load the url page directly in the account creation mode by changing (or adding) the `prompt` url parameter to `create`.

@sandhose does it make sense to you?

## Motivation and context

Improve user experience with MAS and closes #3627

## Screenshots / GIFs

|Before|After|
|-|-|
|<video width="98" alt="image" src="https://github.com/user-attachments/assets/ff90a0aa-a776-43fd-ad15-eb2160c91e78" />|<video width="98" alt="image" src="https://github.com/user-attachments/assets/e7d8f605-9f18-45d6-bd29-69ee597f2272"/>|

## Tests

- Try creating an account on a server with MAS support, such as beta.matrix.org and observe that you're having a wonderful navigation experience.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: libraries/androidutils/src/main/kotlin/io/element/android/libraries/androidutils/uri/UriExtensions.kt [hunk 1]
```diff
@@ -12,3 +12,16 @@ import android.net.Uri
...
+    return buildUpon().apply {
+        clearQuery()
+        existingParams.forEach { existingKey ->
+            if (existingKey != key) {
+                appendQueryParameter(existingKey, getQueryParameter(existingKey))
+            }
+        }
+        appendQueryParameter(key, value)
+    }.build()
+}
```
Comment (bmarty) [inline libraries/androidutils/src/main/kotlin/io/element/android/libraries/androidutils/uri/UriExtensions.kt:27]: Only `appendQueryParameter` exists in `Uri.Builder`, so sadly, I had to create this extension.

**Review by sandhose — COMMENTED** [review]
This is right, although I'm surprised the SDK doesn't surface this parameter?

There is a setter here: https://github.com/matrix-org/matrix-rust-sdk/blob/9c6413551cd6e15f01453e57178fe98e0d12200b/crates/matrix-sdk/src/oidc/auth_code_builder.rs#L77-L80

Which is being used here: https://github.com/matrix-org/matrix-rust-sdk/blob/9c6413551cd6e15f01453e57178fe98e0d12200b/crates/matrix-sdk/src/oidc/mod.rs#L473

And exposed in the bindings here: https://github.com/matrix-org/matrix-rust-sdk/blob/9c6413551cd6e15f01453e57178fe98e0d12200b/bindings/matrix-sdk-ffi/src/client.rs#L367

I feel like there should be an `url_for_oidc_registration` equivalent in the SDK?

**Comment by bmarty** [comment]
OK so maybe we could improve the existing Rust API, to replace `urlForOidcLogin()` by `urlForOidc(prompt: String)`, or simply add an API `urlForOidcRegister()`

@jmartinesp do you think this would be possible?

**Comment by jmartinesp** [comment]
From a very quick look at the code, it seems feasible, although I'd rather make the `Prompt` an enum in the FFI layer so the client can't just pass anything. What do you think?

**Comment by bmarty** [comment]
Yes, an enum would be better than a String. Thanks!

**Comment by bmarty** [comment]
Closing this we will get a better way to handle this thanks to Jorge's work in https://github.com/matrix-org/matrix-rust-sdk/pull/4131.

**Pull request closed without merging**